collector/connectors/odigosrouterconnector/connector.go

```diff
@@ -0,0 +1,373 @@
+package odigosrouterconnector
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	collectorpipeline "go.opentelemetry.io/collector/pipeline"
+	semconv1_26 "go.opentelemetry.io/otel/semconv/v1.26.0"
+	"go.uber.org/zap"
+
+	"github.com/odigos-io/odigos/common"
+	"github.com/odigos-io/odigos/common/consts"
+)
+
+type tracesConfig struct {
+	consumers   connector.TracesRouterAndConsumer
+	defaultCons consumer.Traces
+	logger      *zap.Logger
+}
+
+type metricsConfig struct {
+	consumers   connector.MetricsRouterAndConsumer
+	defaultCons consumer.Metrics
+	logger      *zap.Logger
+}
+
+type logsConfig struct {
+	consumers   connector.LogsRouterAndConsumer
+	defaultCons consumer.Logs
+	logger      *zap.Logger
+}
```

Comment:
nit: can we save the `logger` once on the `routerConnector` instead of each Config?